### DISCUSSION TITLE
PFEM Q&A

### DISCUSSION THREAD

**Discussion by MiguelARV16** [KratosMultiphysics/Kratos#12793] [Q&A]
Hello.
I'm trying to work with the PFEM application for fluids, but I'm getting this error:

Traceback (most recent call last):
  File "C:\ejemplo_pfem_2D_fluid.gid\MainKratos.py", line 40, in <module>
    analysis_stage_module = importlib.import_module(analysis_stage_module_name)
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\GiD\GiD 17.1.0d\scripts\tohil\python\Lib\importlib\__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1310, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1324, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'KratosMultiphysics.PfemFluidDynamicsApplication'

**Comment by FraJoMen** [comment]
Hi Miguel,

Did you manage to solve the issue with the missing `PfemFluidDynamicsApplication` module? I'm curious to know if you were able to complete your simulation, as I'm currently working on a related topic and facing similar uncertainties.

Before diving deeper into Kratos, I’d like to better understand its current capabilities for PFEM-based simulations. My goal is to model shallow landslides or debris flows using PFEM (particularly with large soil deformations and possible fluid–soil interaction), but I want to assess the current viability of this approach within Kratos.

Specifically, I’d like to know:

1. Whether `PfemApplication` and `PfemFluidDynamicsApplication` are actively supported and usable for geomechanical simulations (e.g., slope failures, saturated soil collapse).
2. What constitutive models are currently available and compatible with such simulations.
3. Are there any working examples or recommended Python workflows to set up coupled PFEM simulations involving both fluids and solids?

If anyone has updates or insights on the current state of PFEM in Kratos, I’d really appreciate it. Thanks in advance!

Best regards,  
Francisco

**Reply by rubenzorrilla** [reply]
@AFranci is the right person to answer all your queries.

**Comment by rubenzorrilla** [comment]
> Hello. I'm trying to work with the PFEM application for fluids, but I'm getting this error:
> 
> Traceback (most recent call last): File "C:\ejemplo_pfem_2D_fluid.gid\MainKratos.py", line 40, in analysis_stage_module = importlib.import_module(analysis_stage_module_name) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "C:\Program Files\GiD\GiD 17.1.0d\scripts\tohil\python\Lib\importlib__init__.py", line 90, in import_module return _bootstrap._gcd_import(name[level:], package, level) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "", line 1387, in _gcd_import File "", line 1360, in _find_and_load File "", line 1310, in _find_and_load_unlocked File "", line 488, in _call_with_frames_removed File "", line 1387, in _gcd_import File "", line 1360, in _find_and_load File "", line 1324, in _find_and_load_unlocked ModuleNotFoundError: No module named 'KratosMultiphysics.PfemFluidDynamicsApplication'

Per regards this. It seems as easy as not having `PfemFluidDynamicsApplication` compiled.

**Comment by AFranci** [comment]
Dear Francisco, 
You can email me to introduce yourself and provide more details about the problem you want to simulate. 
I will be happy to answer.
Regards,
Alessandro
